### PR TITLE
[Cleanup] Change Byte to Bool for training masks

### PR DESCRIPTION
**Pull request by JakeStevens**
## Description
Using ByteTensor for indexing has been deprecated in PyTorch, leading to many warnings in the example traning scripts. I've updated these to use BoolTensor.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by yzh119** [comment]
Thanks for your contribution!
Could you please also mention the minimum version number of PyTorch that supports `BoolTensor` in the README file of these examples? Because users who did not upgrade their PyTorch would receive errors after your changes.

**Comment by JakeStevens** [comment]
I've added a check instead, so that if the version of torch being used has BoolTensors, they are used, thus avoiding the warning. Otherwise, ByteTensor is used as in the original code. This way, users who did not upgrade will not get errors since the code is unchanged for them.

**Comment by jermainewang** [comment]
Thanks @JakeStevens for the contribution.
